### ISSUE TITLE
[Wpf] Application.OnTerminating never being called from a derived class

### ISSUE THREAD

**Issue by wildj79** [picoe/Eto#867]
Poking around at the `Eto.Wpf` source code, it looks like this might be intentional.  At any rate, if you derive a class from `Eto.Forms.Application` and try to override `OnTerminating(CancelEventArgs e)`, the override is never called. I've tried working around this by overriding the `OnClosing(CancelEventArgs e)` method in a form, but this causes an `InvalidOperationException` in the `Application` class when you try to cancel the event. 

```
System.InvalidOperationException occurred
  HResult=0x80131509
  Message=Cannot set Visibility or call Show, ShowDialog, or WindowInteropHelper.EnsureHandle after a Window has closed.
```

**Comment by cwensley** [comment]
Thanks for the error report!

**Comment by wildj79** [comment]
@cwensley I'll update my project and give it a try.

**Comment by cwensley** [comment]
Closing as I think this issue was resolved in 2.4.1.  Feel free to comment or reopen if you are still seeing an issue with the latest develop version.

Thanks for submitting the issue!
